crates/sb_core/js/bootstrap.js

```diff
@@ -495,6 +499,7 @@ globalThis.bootstrapSBEdge = (opts, ctx) => {
 		"beforeunload",
 		"unload",
 		"unhandledrejection",
+		"drain",
```

Comment:
What's the difference between `beforeunload` and `drain` event? When should a user function listen to drain over beforeunload?

Is this an event triggered in Deno runtime as well?

Reply:
> Is this an event triggered in Deno runtime as well?
No, it's an edge-runtime specific event.

If the resource exceeds the soft limit, the worker is marked as the retired state, and the worker pool no longer relays new requests to this worker, but the request queue may still contain requests.

When all remaining requests have been processed, the drain event is triggered. This is used to safely remove the task of the http listener from the event loop. This is a different event from `beforeunload`.

@laktek

Reply:
Is it expected to be used by user functions? It feels more like an internal event if it only used for a specific purpose.

Reply:
Yeah, basically it has a purpose to be used internally.

I could have replaced it with a direct function call instead of firing an event, but I just implemented it as an event firing, thought some users might use it.

Reply:
I'm a bit concerned it may be confusing for users - when to use `drain` vs `beforeunload` and the order of them firing. Can an event handler on `drain` block the unload of the worker?

Reply:
The threshold for the beforeunload trigger is specified by the cli flag. (The default is 90%)

`drain` indicates that there will be no more requests, and `beforeunload` indicates that the worker will soon be unloaded because the resource has exceeded the threshold specified by the cli flag.

> Can an event handler on drain block the unload of the worker?

All events will be monitored by the supervisor to prevent cheating by user scripts.
Even if an event handler attempts to prevent termination, it will be defeated by a call to v8's terminate_execution API. This is true for all events triggered by the runtime.

Reply:
ok sounds good! We'll probably have to add docs later for different events to make it clear for users.